web/src/test/java/org/openmrs/web/filter/update/util/CurrentUsersTest.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.web.filter.update.util;
+
+import java.util.List;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.openmrs.User;
+import org.openmrs.api.UserService;
+import org.openmrs.api.context.Context;
+import org.openmrs.web.filter.util.CurrentUsers;
+import org.openmrs.web.test.BaseWebContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mock.web.MockHttpSession;
+
+public class CurrentUsersTest extends BaseWebContextSensitiveTest {
+
+	private static final String USER_SET = "org/openmrs/CurrentUserTest.xml";
+	@Autowired
+	UserService userService;
+
+	@Test
+	
+	public void getCurrentUsernames_shoulReturnUserNamesForLoggedInUsers() {
+		executeDataSet(USER_SET);
+		MockHttpSession session = new MockHttpSession();
+		User user = userService.getUser(5508);
+		Context.authenticate(user.getUsername(),"User12345");
+		Assert.assertEquals(Context.getAuthenticatedUser().getUsername(),user.getUsername());
+		User loggedInUser = Context.getAuthenticatedUser();
+		CurrentUsers.addUser(session,loggedInUser);
```

Comment:
its at line 38

Reply:
In which class?

Reply:
> In which class?

From CurrentUsers class ie  org.openmrs.web.filter.util.CurrentUsers;

Reply:
So CurrentUsers calls itself CurrentUsers.addUser() on line 38?

Reply:
> So CurrentUsers calls itself CurrentUsers.addUser() on line 38?

sorry i am failing to understand this!!

Reply:
Run the application and then put a break point in CurrentUsers.addUser() and tell me if the debugger breaks there at any one time during the running of the application.

Reply:
> Run the application and then put a break point in CurrentUsers.addUser() and tell me if the debugger breaks there at any one time during the running of the application.

ok let me get back here after doing that!

Reply:
> Run the application and then put a break point in CurrentUsers.addUser() and tell me if the debugger breaks there at any one time during the running of the application.

@dkayiwa added a breaking point at CurrentUsers.addUser()  and the debugger never breaks at any time  https://pastebin.com/ehgmYq5G

Reply:
That is exactly what i wanted you to see. And that is, your class is never used anywhere in the running web application.

Reply:
> your class is never used anywhere in the running web application.

@dkayiwa  does this mean we completely remove this class

Reply:
@HerbertYiga no. The point am making is that if you just drop the class in core and you do not make core fill it up with logged in users, then you have not met the goals of the ticket.

Reply:
> The point am making is that if you just drop the class in core and you do not make core fill it with logged in users, then you have not met the goals of the ticket.

thanks @dkayiwa  looking at these,i see i add the logged in users after picking them using Context.getAuthenticatedUser()

User loggedInUser = Context.getAuthenticatedUser();
CurrentUsers.addUser(session,loggedInUser);

Reply:
You did that in a test. But does the test execute at runtime?